test/integration/thesis_test.rb

```diff
@@ -167,6 +167,57 @@ def teardown
     assert_equal new_thesis_path, path
   end
 
+  test 'a confirmation email is sent on form submit' do
+    user = users(:basic)
+    thesis = theses(:two)
+
+    mock_auth(user)
+
+    ClimateControl.modify DISABLE_ALL_EMAIL: 'false' do
+      email = ReceiptMailer.receipt_email(thesis, user)
+
+      assert_emails 1 do
+        patch thesis_path(thesis), params: { thesis: thesis.serializable_hash }
+      end
```

Comment:
Something about this block is tripping me up - I think because the `email = ...` on line 177 doesn't appear to relate to the `assert_emails` block from 179-181. However, if I comment it out, then the assertions on 182 & 183 fail.

Could this be slightly re-organized to move 177 down below the assert_email block?  Or is it actually required in some way that I'm missing? I approve of having tests to verify attributes of the email that gets sent, which I think is the intent here - but it feels like things are interleaved here in way that could be avoided.

Reply:
Totally agree. I think moving the `email` variable assignment next to the relevant assertions makes sense, and now that I'm considering it further, we really only need to verify attributes in the test in which we confirm the addressee. (The basic structure of the email is already verified in the receipt mailer tests.)